File: MedApp/Models/Cita.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MedApp.Models
{
    [JsonObject(IsReference = true)]
    public class Cita : Entidad
    {

        public int PacienteID { get; set; }
        public DateTime Fecha { get; set; }
        public int TipoCitaID { get; set; }

        public virtual Paciente Paciente { get; set; }
        public virtual TipoCita TipoCita { get; set; }
    }
}

File: MedApp/Client/src/app/models/cita.ts
import { Entidad } from './entidad';
import { Paciente } from './paciente';
import { TipoCita } from './tipo-cita';

export interface Cita extends Entidad {
    Fecha:Date;
    PacienteID: number;
    TipoCitaID: number;
    Paciente?: Paciente;
    TipoCita?: TipoCita;
}
